### PR TITLE
test: fix typo

### PR DESCRIPTION
**Pull request by gangliao**

### CODE REVIEW

**Comment by nevi-me** [comment]
Hi @gangliao , may you please kindly provide some context.

File: rust/arrow/src/ipc/convert.rs [hunk 1]
```diff
@@ -859,7 +859,7 @@ mod tests {
             0, 0,
         ];
         let ipc2 = ipc::root_as_message(&bytes[..]).unwrap();
-        let schema2 = ipc.header_as_schema().unwrap();
+        let schema2 = ipc2.header_as_schema().unwrap();
```
Comment (alamb) [inline rust/arrow/src/ipc/convert.rs:862]: reading the code, it does appear that the intent of this test is to compare the schema from two different IPC message (aka `schema2` should be coming from the `ipc2` message) as @gangliao  has done here. 


The test seems to have failed however https://github.com/apache/arrow/pull/9764/checks?check_run_id=2161212421


```
thread 'ipc::convert::tests::schema_from_bytes' panicked at 'assertion failed: `(left == right)`
  left: `Schema { endianness: Little, fields: Some([Field { name: Some("field1"), nullable: false, type_type: Int, type_: Int { bitWidth: 32, is_signed: false }, dictionary: None, children: Some([]), custom_metadata: None }]), custom_metadata: None, features: None }`,
 right: `Schema { endianness: Little, fields: Some([Field { name: Some("field1"), nullable: false, type_type: Int, type_: Int { bitWidth: 32, is_signed: false }, dictionary: None, children: None, custom_metadata: None }]), custom_metadata: Some([]), features: None }`', arrow/src/ipc/convert.rs:864:9
```

The difference is that `schema` has `children: Some([])` and `schema2` has `children: None`

Comment (gangliao) [reply rust/arrow/src/ipc/convert.rs:862]: yes, I didn't find a good way to fix it.

Comment (alamb) [reply rust/arrow/src/ipc/convert.rs:862]: maybe we can assert the individual fields that are the same. Something like

```
assert_eq!(schema.endianess, schema2.endianess)
assert_eq!(schema.fields(), schema2.fields())
...
```

**Comment by alamb** [comment]
I filed https://github.com/apache/arrow-rs/issues/250 in arrow-rs to track this issue. Thanks again @gangliao  for finding it and submitting a PR

https://github.com/apache/arrow/pull/10096 has removed the arrow implementation from this repository (it now resides in https://github.com/apache/arrow-rs and https://github.com/apache/arrow-datafusion) in the hopes of streamlining the development process

Please re-target this PR (let us know if you need help doing so) to one/both of the new repositories. 

Thank you for understanding and helping to make arrow-rs and datafusion better

**Pull request closed without merging**